packages/blocks/src/api/parser/index.js

```diff
@@ -124,20 +132,14 @@ function createMissingBlockType( rawBlock ) {
 		isCommentDelimited: false,
 	} );
 
-	// Preserve full block content for use by the unregistered type
-	// handler, block boundaries included.
-	const originalContent = serializeRawBlock( rawBlock, {
-		isCommentDelimited: true,
-	} );
-
 	return {
 		blockName: unregisteredFallbackBlock,
 		attrs: {
 			originalName: rawBlock.blockName,
-			originalContent,
+			originalContent: rawBlock.sourceMarkup,
 			originalUndelimitedContent,
 		},
-		innerHTML: rawBlock.blockName ? originalContent : rawBlock.innerHTML,
```

Comment:
What's the reasoning for the removal of the condition here? Is it something that never happens?

Reply:
previously we have been re-generating block output from the parsed inputs. this is the reason that the editor can silently corrupt or lose content. for HTML soup outside of the blocks though there's no way to re-generate the block since there is no block. because of this we special-case the HTML soup and use `innerHTML` directly, which ironically is closer to the correct behavior for regular blocks too.

after preserving the source markup every parsed block contains that original content which we can thread through and it doesn't matter if it's a block proper or not, so no more need for the conditional and special-casing.